sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -932,8 +934,17 @@ class Analyzer(
             case v1Table: V1Table =>
               v1SessionCatalog.getRelation(v1Table.v1Table)
             case table =>
+              // Note that SessionCatalogAndIdentifier doesn't update the namespace if the resolved
+              // identifier has a single-part name because it could reference a temp view which does
+              // not belong to any namespaces. For v1 tables, namespace is resolved in
+              // `SessionCatalog.getRelation`.
+              val ns = if (ident.namespace.isEmpty) {
+                catalogManager.currentNamespace
```

Comment:
What if the namespace is really `[]`?

Reply:
I'm fixing it in https://github.com/apache/spark/pull/27550

Reply:
Good catch! I will take a look at the PR. Thanks!

Reply:
This is still problematic even after #27550 because we allow `spark_catalog.t`. It will be resolved to an identifier with an empty namespace in `CatalogAndIdentifier` whereas `v1SessionCatalog` always uses current database if the given namespace (database) is empty. Should I just go ahead and disallow `spark_catalog.t`? What do you think? We briefly discussed this issue here: https://github.com/apache/spark/pull/27550#discussion_r379245778

Reply:
yea let's do it!

Reply:
will do!

Reply:
@cloud-fan this is now fixed without this hack.